services/ingestion/etl/just/csv/src/lib/transform.js

```diff
@@ -0,0 +1,166 @@
+              parseFloat(projectLongitude) || 0,
+              parseFloat(projectLatitude) || 0,
+            ],
+          }
+        : null,
+    },
+  ];
+};
+
+/**
+ * Preprocess related links
+ *
+ * Depends on record['Related links'] field
+ *
+ * @memberof AgriCsvTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array|Object} List of {RelatedLink}
+ *
+ * @example
+ * input => "<a href=\"https://ec.europa.eu/inea/en/ten-t/ten-t-projects/projects-by-country/multi-country/2013-eu-92069-s\">INEA</a>;<a href=\"https://europa.eu/investeu/projects/central-european-green-corridors_en\">InvestEU</a>"
+ * output => [
+ *    { label: "INEA", url: "https://ec.europa.eu/inea/en/ten-t/ten-t-projects/projects-by-country/multi-country/2013-eu-92069-s" }
+ *    { label: "InvestEU", url: "https://europa.eu/investeu/projects/central-european-green-corridors_en" }
+ *  ]
+ */
+const getRelatedLinks = record =>
+  (record.field_prj_link || '')
+    .split(';')
+    .filter(link => link)
+    .map(link => {
+      const matches = link.match(/<a .*href="(.*)".*>(.*)<\/a>/i);
+
+      if (Array.isArray(matches) && matches.length === 3) {
+        return {
+          url: matches[1],
+          label: matches[2],
+        };
+      }
+
+      return {
+        url: '',
+        label: '',
+      };
+    })
+    .filter(link => link !== null);
+
+/**
+ * Map fields for JUST producer, CSV file types
+ *
+ * Example input data: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/just/csv/test/stubs/record.json|stub}
+ *
+ * Transform function: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/just/csv/src/lib/transform.js|implementation details}
+ * @name JustCsvTransform
+ * @param {Object} record Piece of data to transform before going to harmonized storage.
+ * @returns {Project} JSON matching the type fields.
+ */
+export default (record: Object): Project => {
+  // Preprocess budget
+  const budgetObject = {
+    total_cost: {
+      value: Number(record.field_prj_total_budget) || 0,
+      currency: '',
+      raw: record.field_prj_total_budget || '',
+    },
+    eu_contrib: {
+      value: Number(record.field_prj_eu_budget) || 0,
+      currency: 'EUR',
+      raw: record.field_prj_eu_budget || '',
+    },
+    private_fund: { value: 0, currency: '', raw: '' },
+    public_fund: { value: 0, currency: '', raw: '' },
+    other_contrib: { value: 0, currency: '', raw: '' },
+    funding_area: getFundingArea(),
+    mmf_heading: '',
+  };
+
+  // Preprocess locations
+  const locationArray = getLocations(record);
+
+  // Preprocess related links
+  const links = getRelatedLinks(record);
+
+  // Preprocess results
+  const resultObject = {
+    available: '',
+    result: record.Results || '',
+  };
+
+  // Map the fields
+  return {
+    action: record.field_prj_cat_actions,
+    budget: budgetObject,
+    call_year: record.field_prj_year,
```

Comment:
Please provide checks (or default) for all fields like this which could be missing.